python/foglamp/services/core/server.py

```diff
@@ -771,10 +774,41 @@ def start(cls):
                 try:
                     response = json.loads(result)
                     response['message']
-                    _logger.info("Successfully shut down the %s service %s.", svc._type, svc._name)
+                    _logger.info("Shutdown scheduled for %s service %s at %s", svc._type, svc._name, response['message'])
                 except KeyError:
                     raise
 
+    @classmethod
+    async def poll_microservices_unregister(cls):
+        """ poll microservice shutdown endpoint for non core micro-services"""
+        try:
+            shutdown_threshold = 0
+            found_services = ServiceRegistry.get()
+            cls._service_shutdown_threshold = 5 * (len(found_services) - 2)
+            while True:
+                services_to_stop = list()
+                for fs in found_services:
+                    if fs._name in ("FogLAMP Storage", "FogLAMP Core"):
+                        continue
+                    if fs._status not in [ServiceRecord.Status.Running, ServiceRecord.Status.Unresponsive]:
+                        continue
+                    services_to_stop.append(fs)
+                if len(services_to_stop) == 0:
+                    _logger.info("All microservices, except Core and Storage, have been shutdown.")
+                    return
+                shutdown_threshold += 1
```

Comment:
increase by 5?

Reply:
1 to match the sleep seconds. 1 is better as it avoids unnecessary waiting.

Reply:
but while will consume more resources here. and we need them for services to cancel their task and clean up.